lib/rubocop/cop/sorbet/signatures/signature_build_order.rb

```diff
@@ -31,34 +30,22 @@ module Sorbet
       class SignatureBuildOrder < ::RuboCop::Cop::Cop # rubocop:todo InternalAffairs/InheritDeprecatedCopClass
         include SignatureHelp
 
-        ORDER =
-          [
-            :abstract,
-            :override,
-            :overridable,
-            :type_parameters,
-            :params,
-            :returns,
-            :void,
-            :soft,
-            :checked,
-            :on_failure,
-          ].each_with_index.to_h.freeze
-
         # @!method root_call(node)
         def_node_search(:root_call, <<~PATTERN)
-          (send nil? {#{ORDER.keys.map(&:inspect).join(" ")}} ...)
+          (send nil? #builder? ...)
         PATTERN
 
         def on_signature(node)
           calls = call_chain(node.children[2]).map(&:method_name)
           return if calls.empty?
 
-          # While the developer is typing, we may have an incomplete call statement, which means `ORDER[call]` will
-          # return `nil`. In that case, invoking `sort_by` will raise
-          return if calls.any? { |call| ORDER[call].nil? }
-
-          expected_order = calls.sort_by { |call| ORDER[call] }
+          expected_order = calls.sort_by do |call|
+            builder_method_indexes.fetch(call) do
+              # Abort if we don't have a configured order for this call,
+              # likely because the method name is still being typed.
+              return nil
+            end
+          end
```

Comment:
I refactored this to perform the check during the same loop as generating the sort keys, and tried to make the comment more focused on the impact than the implementation.

Interestingly, having just `return` without `nil` here in a block is an offense.

Reply:
This code has been refactored, so this is no longer relevant.